api/paidAction/index.js

```diff
@@ -45,6 +45,9 @@ export default async function performPaidAction (actionType, args, context) {
     context.me = me ? await models.user.findUnique({ where: { id: me.id } }) : undefined
     context.cost = await paidAction.getCost(args, context)
 
+    context.sybilFeePercent = paidAction.getSybilFeePercent ? await paidAction.getSybilFeePercent(args, context) : undefined
+    if (context.sybilFeePercent != null && context.sybilFeePercent < 0n) throw new Error('sybil fee percent cannot be negative')
```

Comment:
I don't like `!=`, it always looks like a bug in JS. Try to use `!==`, `!` (infix operator) or optional chaining for cases like this.

For example, in this case, you could use `context?.sybilFeePercent < 0n` since comparing a number to `null` or `undefined` always returns `false`:

```
$ node
Welcome to Node.js v20.18.0.
Type ".help" for more information.
> null < 0n
false
> null > 0n
false
> undefined < 0n
false
> undefined > 0n
false
```

But I don't know if _that_ looks like a bug to other developers.

Also, if `sybilFeePercent` is not defined, shouldn't we apply the default instead of throwing?

Reply:
I disagree that  `!= null` looks like a bug.  It has a [defined behavior ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality ) and this is one of the correct use cases. Since the code is checking for both null and undefined.

This is  a safety check that throws if the sybil fee is negative. null or undefined fees are fine, in this case the default will be use in wrapInvoice.

Reply:
JS defined behavior is usually a foot gun and when I see `!=`, it's a bug in 99% of the cases in my experience (and that's why you should look out for them, too). I know considering least common denominators between devs is annoying but it will also be annoying to always pause and wonder: "did whoever wrote this code knew what they were doing?"

What's wrong with `context?.sybilFeePercent < 0n`? Can we get behind that?